platforms/chibios/bootloaders/stm32duino.c

```diff
@@ -18,4 +18,5 @@
 
 #include <ch.h>
 
-__attribute__((weak)) void bootloader_jump(void) { NVIC_SystemReset(); }
+__attribute__((weak)) void bootloader_jump(void) { keyboard_jump(); }
+__attribute__((weak)) void keyboard_jump(void) { NVIC_SystemReset(); }
```

Comment:
For stm32duino the bootloader handling in QMK is somewhat weird — the bootloader magic word is in the RTC backup RAM, and it gets set to the “enter the bootloader” value in the QMK startup code (so that pressing Reset just once after the QMK firmware had started will result in entering the bootloader for the unlimited time). Here `keyboard_jump()` may need to clear that magic word, so that the keyboard restart will actually work without getting stuck in the bootloader (especially if the “fastboot” version of the bootloader is used).